sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/join/CoGbkResult.java

```diff
@@ -422,7 +422,7 @@ private UnionValueIterator(int tag, Iterator<RawUnionValue> unions, Boolean[] co
 
     @Override
     public boolean hasNext() {
-      if (containsTag[tag] == Boolean.FALSE) {
+      if (Boolean.FALSE.equals(containsTag[tag])) {
```

Comment:
`null` is a valid value in `containsTag`, see [CoGbkResult.java#L124](https://github.com/swegner/incubator-beam/blob/3875071b5289bb729ad84de97b161d679d3c3292/sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/join/CoGbkResult.java#L124): 

```java
// This is a trinary-state array recording whether a given tag is present in the tail. The
// initial value is null (unknown) for all tags, and the first iteration through the entire
// list will set these values to true or false to avoid needlessly iterating if filtering
// against a given tag would not match anything.
final Boolean[] containsTag = new Boolean[schema.size()];
```

Dereferencing for `#booleanValue()` may throw.